openvdb/points/AttributeSet.h

```diff
@@ -540,6 +558,82 @@ class OPENVDB_API AttributeSet::Descriptor
 #endif
 }; // class Descriptor
 
+
+////////////////////////////////////////
+
+/// @brief A convenience class that collates the Descriptor with metadata that
+/// is only available on each AttributeArray, such as whether the array is
+/// uniform or not. This is useful for querying and during construction of new
+/// AttributeSets.
+/// @note For performance reasons, this class uses copy-on-write for storing
+/// the Descriptor so any modification of the source Descriptor also requires
+/// a deep-copy for the data in this class to remain valid.
+class OPENVDB_API AttributeSet::Info {
+public:
+    struct OPENVDB_API Array {
+        Index stride = 1;
+        bool constantStride = true;
+        bool hidden = false;
+        bool transient = false;
+        bool group = false;
+        bool string = false;
+    }; // struct Array
+
+    /// Default constructor
+    Info() = default;
+
+    /// Construct a new Info class from the given Descriptor.
+    /// @param descriptorPtr the descriptor
+    /// @note Array metadata is default-initialized when using this constructor
+    explicit Info(const AttributeSet::Descriptor::Ptr& descriptorPtr);
+
+    /// Construct a new Info class from the given AttributeSet.
+    /// @param attributeSet the attribute set
+    /// @details This method extracts and stores Array metadata from the
+    /// AttributeArray objects stored within it
+    explicit Info(const AttributeSet& attributeSet);
+
+    /// @brief Return the descriptor
+    Descriptor& descriptor() { return *mDescriptor; }
+    /// @brief Return the descriptor
+    const Descriptor& descriptor() const { return *mDescriptor; }
+    /// @brief Return a shared pointer to the descriptor
+    Descriptor::Ptr descriptorPtr() const { return mDescriptor; }
+    /// @brief Return the number of array attributes
+    size_t size() const { return mDescriptor->size(); }
+
+    /// @brief Return an Array metadata object by name
+    /// @param name The name of the array
+    /// @throws LookupError if name does not exist
+    Array& arrayInfo(const Name& name);
```

Comment:
What's the purpose of returning modifiable Array objects with any of these member methods?

Reply:
You can use it in combination with the `AttributeSet(const Info& ...)` constructor to compose exactly how you'd like to construct the desired AttributeSet. I should highlight that there's no link between this `Info` data and the member descriptor. It's purely transitory metadata that is passed down to the constructed AttributeArray.